src/components/client/one-time-donation/steps/FirstStep.tsx

```diff
@@ -343,9 +424,74 @@ export default function FirstStep() {
             <FormTextField
               name="otherAmount"
               type="number"
+              value={formik.values.otherAmount === 1 ? '' : formik.values.otherAmount}
+              defaultValue={''}
               label={t('first-step.amount')}
+              lang={i18n.language}
+              onKeyDown={(e) => {
+                if (
+                  formik.errors.otherAmount &&
+                  e.key !== 'Backspace' &&
+                  e.key !== 'Delete' &&
+                  !isInteger(formik.values.otherAmount)
+                ) {
+                  e.preventDefault()
+                  return
+                }
+                if (decimalSeparator !== e.key && (e.key === '.' || e.key === ',')) {
+                  e.preventDefault()
+                  return
+                }
+
+                if (
+                  (e.key.charCodeAt(0) >= 48 && e.key.charCodeAt(0) <= 57) ||
```

Comment:
looks like repeated code from the onKeyDown above, so definitely needs to be a reusable function